File: src/main/java/com/puertobahia/iceberg/service/PerfilService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.puertobahia.iceberg.service;


import com.puertobahia.iceberg.entity.Perfil;
import java.util.List;

/**
 *
 * @author Full Stack JavaScrip
 */
public interface PerfilService {

    public List<Perfil> getAllPerfil();

    public Perfil getById(Long id);

    public void saveOrUpdate(Perfil perfil);

    public void delete(Long id);
}

File: src/main/webapp/resources/assets/js/app/services/perfilService.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

var servicesModule = angular.module('AppServices');

servicesModule.factory('perfilService', ['$http', function ($http) {
    return {
        apiUrl: apiUrl,
        getAllPerfil: function () {
            return $http.get(this.apiUrl + 'perfil/');
        },
        getPerfilById: function (perfilId) {
            return $http.get(this.apiUrl + 'perfil/' + perfilId);
        },
        createPerfil: function (perfil) {
            return $http.post(this.apiUrl + 'perfil/', perfil);
        },
        updatePerfil: function (perfil) {
            return $http.put(this.apiUrl + 'perfil/', perfil);
        },
        deletePerfil: function (perfilId) {
            return $http.delete(this.apiUrl + 'perfil/' + perfilId);
        }
    };
    }]);
